### PR TITLE
Numbers: Removed the redundant checks of igcd method and simplified it

### LINKED ISSUE #15839: Simplify the igcd method
In #15837 there was a suggestion in a review comment to simplify the `igcd`method (which wasn't really part of that pull request). Hence, there is a more or less ready solution that should be tested etc.

Although, I think the logic of `igcd` can be simplified a bit. The code seems to have some redundant checks:

https://github.com/sympy/sympy/blob/11e57159cb8bcaa1a75e1bf2119430f63d2f3815/sympy/core/numbers.py#L179-L201

The above code can be reduced to something like:
```
    if 1 in args:
        a = 1
        all(as_int(arg) for arg in args)
    else:
        a = abs(as_int(args[0]))
        for arg in args[1:]:
            b = as_int(arg)
            if b == 0:
                continue
            if b < 0:
                b = -b
            a = igcd2(a, b)
    return a
```

_Originally posted by @sidhantnagpal in https://github.com/sympy/sympy/pull/15837#issuecomment-457292210_

### PR DESCRIPTION
**Pull request by supreet11agrawal**
In igcd method, various redundant checks are present such as in line 185 a is surely not equal to 1.
Also in line 192, b != 1 as we have checked that earlier in line 179. Many other such redundancies have been removed.
I have done this as advised by @oscargus and ran all the checks.

#### References to other Issues or PRs

Fixes #15839 

#### Brief description of what is fixed or changed
igcd method is simplified and redundant checks have been removed. All the tests are running. Kindly review and tell me if I should change something @oscargus @asmeurer 

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/core/numbers.py [hunk 1]
```diff
@@ -179,26 +179,16 @@ def igcd(*args):
...
-            if b == 1:
-                a = 1
-                break
             if b < 0:
                 b = -b
-            a = igcd2(a, b)
-    while k < len(args):
-        ok = as_int(args[k])
-        k += 1
+            a = igcd2(a,b)
```
Comment (jksuom) [inline sympy/core/numbers.py]: Please add a space after comma.

Comment (supreet11agrawal) [reply sympy/core/numbers.py]: ok

**Comment by oscargus** [comment]
Looks good to me! (But I have no merge access.)

**Comment by jksuom** [comment]
Would it make sense to start by applying `to_int` to all arguments?

**Comment by supreet11agrawal** [comment]
> Would it make sense to start by applying `to_int` to all arguments?

No, I don't think that will be good as igcd should give an error when input has non-integer arguments

**Comment by jksuom** [comment]
`as_int` will raise ValueError for a non-integer argument.

**Comment by smichr** [comment]
I like the idea of converting all to ints from the start:

```python
args = [as_int(i) for i in args]
if 1 in args:
    return 1
a = args.pop()
for b in args:
    a = igcd2(a, b if b > 0 else -b) if b else a
```

**Comment by jksuom** [comment]
We could also start with

    args = [abs(as_int(i)) for i in args]

**Comment by supreet11agrawal** [comment]
Maybe it will work. @jksuom  I think you should try implementing this and check the tests

**Comment by supreet11agrawal** [comment]
@asmeurer Can you review this one

**Comment by smichr** [comment]
> line won't work bcz we can't change the list

It's no problem to pop from a list. If it bothers you to alter the input argument -- in this case, the routine is so short that I don't think it matters -- then you could just use `ints = [abs(as_int(i)) for i in args]`
```python
ints = [abs(as_int(i)) for i in args]
if 1 in ints:
    return 1  # S.One or 1, not sure in this context
a = ints.pop()
for b in ints:
    a = igcd2(a, b) if b else a
return a  # or S(a), not sure in this context
```

**Comment by smichr** [comment]
...or simply,
```python
ints = filter(None, [abs(as_int(i)) for i in args])
a = ints.pop() if ints else 1
return reduce(igcd2, ints, a) if 1 not in ints else 1
```

**Comment by oscargus** [comment]
Thanks @asmeurer ! I will probably be cautious when it comes to merging for a while, still think I need a better understanding of the whole project. However, I can definitely help out, especially when it comes to labels, cleaning up in the lists etc.

**Comment by supreet11agrawal** [comment]
@oscargus does something needs to be changed?

**Comment by jksuom** [comment]
I would prefer the version in [this comment](https://github.com/sympy/sympy/pull/15841#issuecomment-457712884). It is quite readable and follows closely to the original. I think that the return values could be `1` and `a` as they are currently.

**Comment by supreet11agrawal** [comment]
> I would prefer the version in [this comment](https://github.com/sympy/sympy/pull/15841#issuecomment-457712884). It is quite readable and follows closely to the original. I think that the return values could be `1` and `a` as they are currently.

Ok, I will do the necessary changes. I am also quite positive that this would work

**Pull request merged**